UnityProject/Assets/Scripts/Systems/Electricity/FunctionsAndClasses/ElectricityFunctions.cs

```diff
@@ -185,76 +185,49 @@ public static float WorkOutResistance(Dictionary<IntrinsicElectronicData, VIRRes
 			return 1 / ResistanceXAll;
 		}
 
-
-		public static Dictionary<IntrinsicElectronicData, float> AnInterestingDictionary =
-			new Dictionary<IntrinsicElectronicData, float>();
-
-		public static void WorkOutActualNumbers(IntrinsicElectronicData ElectricItem)
+		public static void WorkOutActualNumbers(IntrinsicElectronicData electricItem)
 		{
 			//Sometimes gives wrong readings at junctions, Needs to be looked into
-			float Current = 0; //Calculates the actual voltage and current flowing through the Node
-			float Voltage = 0;
-			foreach (var Supply in ElectricItem.Data.SupplyDependent)
+			//Calculates the actual voltage and current flowing through the Node
+			float current = 0;
+			float voltage = 0;
+
+			foreach (var supply in electricItem.Data.SupplyDependent)
 			{
-				Voltage += Supply.Value.SourceVoltage;
+				//Voltages easy to work out just add up all the voltages from different sources
+				voltage += supply.Value.SourceVoltage;
 			}
 
-			lock (AnInterestingDictionary)
+			foreach (var currentIDItem in electricItem.Data.SupplyDependent)
 			{
-				AnInterestingDictionary.Clear(); //Voltages easy to work out just add up all the voltages from different sources
-				foreach (var CurrentIDItem in ElectricItem.Data.SupplyDependent)
+				foreach (var currentItem in currentIDItem.Value.CurrentComingFrom)
 				{
-					foreach (var CurrentItem in CurrentIDItem.Value.CurrentComingFrom)
-					{
-						if (AnInterestingDictionary.ContainsKey(CurrentItem.Key))
-						{
-							AnInterestingDictionary[CurrentItem.Key] += (float) CurrentItem.Value.Current();
-						}
-						else
-						{
-							AnInterestingDictionary[CurrentItem.Key] = (float) CurrentItem.Value.Current();
-						}
-					}
-
-					foreach (var CurrentItem in CurrentIDItem.Value.CurrentGoingTo)
-					{
-						if (AnInterestingDictionary.ContainsKey(CurrentItem.Key))
-						{
-							AnInterestingDictionary[CurrentItem.Key] += (float) -CurrentItem.Value.Current();
-						}
-						else
-						{
-							AnInterestingDictionary[CurrentItem.Key] = (float) -CurrentItem.Value.Current();
-						}
-					}
+					current += (float) currentItem.Value.Current();
 				}
 
-				foreach (var CurrentItem in AnInterestingDictionary)
+				foreach (var currentItem in currentIDItem.Value.CurrentGoingTo)
 				{
-					if (CurrentItem.Value > 0)
```

Comment:
was there any scenario where this was false?  might be worth putting in else and then putting a breakpoint to see if anything triggers it

Reply:
Added a check for it